src/d3d12/d3d12_command_queue.cpp

```diff
@@ -33,9 +33,16 @@ namespace wr::d3d12
 
 		cmd_queue->m_native->ExecuteCommandLists(native_lists.size(), native_lists.data());
 
-		fence->m_fence_value++;
-		Signal(fence, cmd_queue);
+		if (fence != nullptr)
```

Comment:
Otherwise it would crash the program if you don't supply a fence, and sometimes it isn't necessary to synchronize stuff because you're already doing it on the GPU, through things like barriers.